account/account.go

```diff
@@ -45,58 +41,335 @@ func SignBytes(chainID string, o Signable) []byte {
+
+func (caw concreteAccountWrapper) Sequence() uint64 {
+	return caw.ConcreteAccount.Sequence
+}
+
+func (caw concreteAccountWrapper) StorageRoot() []byte {
+	return caw.ConcreteAccount.StorageRoot
+}
+
+func (caw concreteAccountWrapper) Permissions() ptypes.AccountPermissions {
+	return caw.ConcreteAccount.Permissions
+}
+
+func (caw concreteAccountWrapper) Encode() []byte {
+	return caw.ConcreteAccount.Encode()
+}
+
+func (caw concreteAccountWrapper) MarshalJSON() ([]byte, error) {
+	return json.Marshal(caw.ConcreteAccount)
+}
+
+// Account mutation via MutableAccount interface
+var _ MutableAccount = concreteAccountWrapper{}
+
+func (caw concreteAccountWrapper) SetPublicKey(pubKey PublicKey) MutableAccount {
+	caw.ConcreteAccount.PublicKey = pubKey
+	addressFromPubKey, err := AddressFromBytes(pubKey.Address())
+	if err != nil {
+		// We rely on this working in all over the place so shouldn't happen
+		panic(fmt.Errorf("could not obtain address from public key: %v", pubKey))
+	}
+	// We don't want the wrong public key to take control of an account so we panic here
+	if caw.ConcreteAccount.Address != addressFromPubKey {
+		panic(fmt.Errorf("attempt to set public key of account %s to %v, "+
+			"but that public key has address %s",
+			caw.ConcreteAccount.Address, pubKey, addressFromPubKey))
+	}
+	return caw
+}
+
+func (caw concreteAccountWrapper) SubtractFromBalance(amount uint64) MutableAccount {
+	if amount > caw.Balance() {
+		panic(fmt.Errorf("insufficient funds: attempt to subtract %v from the balance of %s",
+			amount, caw.ConcreteAccount))
+	}
+	caw.ConcreteAccount.Balance -= amount
+	return caw
+}
+
+func (caw concreteAccountWrapper) AddToBalance(amount uint64) MutableAccount {
+	if binary.IsUint64SumOverflow(caw.Balance(), amount) {
+		panic(fmt.Errorf("uint64 overflow: attempt to add %v to the balance of %s",
+			amount, caw.ConcreteAccount))
+	}
+	caw.ConcreteAccount.Balance += amount
+	return caw
+}
+
+func (caw concreteAccountWrapper) SetCode(code []byte) MutableAccount {
+	caw.ConcreteAccount.Code = code
+	return caw
+}
+
+func (caw concreteAccountWrapper) IncSequence() MutableAccount {
+	caw.ConcreteAccount.Sequence += 1
+	return caw
+}
+
+func (caw concreteAccountWrapper) SetStorageRoot(storageRoot []byte) MutableAccount {
+	caw.ConcreteAccount.StorageRoot = storageRoot
+	return caw
+}
+
+func (caw concreteAccountWrapper) SetPermissions(permissions ptypes.AccountPermissions) MutableAccount {
+	caw.ConcreteAccount.Permissions = permissions
+	return caw
+}
+
+func (caw concreteAccountWrapper) MutablePermissions() *ptypes.AccountPermissions {
+	return &caw.ConcreteAccount.Permissions
+}
+
+func (caw concreteAccountWrapper) Copy() MutableAccount {
+	return concreteAccountWrapper{caw.ConcreteAccount.Copy()}
+}
+
+// concreteAccount Wrapper
+//----------------------------------------------
+// Encoding/decoding
+
+func AccountEncoder(o interface{}, w io.Writer, n *int, err *error) {
+	wire.WriteBinary(o.(*ConcreteAccount), w, n, err)
```

Comment:
I think you could use the `Account` interface instead of `interface{}` and then do something like this https://gist.github.com/aludvik/12b816a7026850f24c0f5f64463c2902